### PR TITLE
infra proxy list pages UI improvements

### PR DESCRIPTION
**Pull request by vinay033**
### :nut_and_bolt: Description: What code changed, and why?

new UI for the list pages for read-only views (tabs UI, cookbooks, roles, environments, data bags and clients)
### :chains: Related Resources
https://github.com/chef/automate/issues/4663
### :+1: Definition of Done
I have added changes to the new UI list pages.
### :athletic_shoe: How to Build and Test the Change
```
STEP 1
inside the hab studio

[default:/src:0]# build components/automate-ui-devproxy/
[default:/src:0]# start_automate_ui_background
[default:/src:0]# start_all_services

STEP 2
open new window
go to automate UI path

$ cd components/automate-ui
and run the command 

npm run serve:hab

To add data https://github.com/chef/automate/blob/master/dev-docs/adding-data/adding_test_data.md#adding-data-to-infra-views
Go to Infrastructure tab >> Chef Servers it's under Chef-Server feature flag.
```
### :white_check_mark: Checklist

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [ ] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable
clients
Databags
Environments
roles
cookbooks

### CODE REVIEW

**Comment by shaik80** [comment]
can i know what  ui changes we are making, please describe in pr

**Comment by vinay033** [comment]
> can i know what ui changes we are making, please describe in pr

can you please check the related issue I have attached a screenshot.

**Comment by vivekshankar1** [comment]
hi @vinay033, i see the list menu and create databag mentioned in the issue. But there are few text changes as well in the pr. Is there any issue related or may i know what are we referring to for these text changes?

**Comment by vinay033** [comment]
> i see the list menu and create databag mentioned in the issue and those are removed. But there are few text changes as well in the pr. Is there any issue related or may i know what are we referring to for these text changes?

right now we are only targeting the read-only changes.. after the read-only release will add the create/edit/delete.

**Review by vivekshankar1 — APPROVED** [review]
Thank you. The changes look good and are as suggested in the issue screenshot.

File: components/automate-ui/src/app/modules/infra-proxy/data-bags-list/data-bags-list.component.html [hunk 1]
```diff
@@ -2,11 +2,14 @@
   <chef-loading-spinner *ngIf="dataBagsListLoading" size="50"></chef-loading-spinner>
   <app-empty-state *ngIf="authFailure" moduleTitle="data bags" (resetKeyRedirection)="resetKeyTabRedirection($event)"></app-empty-state>
   <ng-container  *ngIf="!dataBagsListLoading && !authFailure">
+    <div class="head-container">
+      <!-- <p>Name of Data bags</p> -->
```
Comment (shaik80) [inline components/automate-ui/src/app/modules/infra-proxy/data-bags-list/data-bags-list.component.html:6]: can you remove this comment

**Review by shaik80 — CHANGES_REQUESTED** [review]

**Comment by msorens** [comment]
@kalroy Why did you mark this PR blocked?

File: components/automate-ui/src/app/modules/infra-proxy/clients/clients.component.html [hunk 2]
```diff
@@ -4,8 +4,8 @@
   <ng-container *ngIf="!clientsListLoading && !authFailure">
     <chef-table>
       <chef-thead>
-        <chef-tr>
-          <chef-th>Clients Name</chef-th>
+        <chef-tr class="no_border_tr">
+          <chef-th class="no_border_th">Name of Clients</chef-th>
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/clients/clients.component.html:8]:
```suggestion
          <chef-th class="no_border_th">Name</chef-th>
```

File: components/automate-ui/src/app/modules/infra-proxy/data-bags-list/data-bags-list.component.html [hunk 3]
```diff
@@ -2,11 +2,14 @@
...
+      <chef-tr class="no_border_tr">
+        <chef-th class="no_border_th">Name of Data bags</chef-th>
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/data-bags-list/data-bags-list.component.html:8]:
```suggestion
        <chef-th class="no_border_th">Name</chef-th>
```

File: components/automate-ui/src/app/modules/infra-proxy/environments/environments.component.html [hunk 4]
```diff
@@ -4,8 +4,8 @@
   <ng-container *ngIf="!environmentsListLoading && !authFailure">
     <chef-table>
       <chef-thead>
-        <chef-tr>
-          <chef-th>Environments Name</chef-th>
+        <chef-tr class="no_border_tr">
+          <chef-th class="no_border_th">Name of Environments</chef-th>
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/environments/environments.component.html:8]:
```suggestion
          <chef-th class="no_border_th">Name</chef-th>
```

File: components/automate-ui/src/app/modules/infra-proxy/cookbooks/cookbooks.component.html [hunk 5]
```diff
@@ -4,9 +4,9 @@
   <ng-container  *ngIf="!cookbooksListLoading && !authFailure">
     <chef-table>
       <chef-thead>
-        <chef-tr>
-          <chef-th>Cookbook Name</chef-th>
-          <chef-th>Cookbook Version</chef-th>
+        <chef-tr class="no_border_tr">
+          <chef-th class="no_border_th">Name of Cookbook</chef-th>
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/cookbooks/cookbooks.component.html:8]:
```suggestion
          <chef-th class="no_border_th">Name</chef-th>
```

File: components/automate-ui/src/app/modules/infra-proxy/infra-roles/infra-roles.component.html [hunk 6]
```diff
@@ -4,10 +4,10 @@
...
     <chef-table>
       <chef-thead>
-        <chef-tr>
-          <chef-th>Role Name</chef-th>
-          <chef-th>Description</chef-th>
-          <chef-th>Environments</chef-th>
+        <chef-tr class="no_border_tr">
+          <chef-th class="no_border_th">Name of Role</chef-th>
+          <chef-th class="no_border_th">Description of Roles</chef-th>
+          <chef-th class="no_border_th">Environments of Roles</chef-th>
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/infra-roles/infra-roles.component.html:10]:
```suggestion
          <chef-th class="no_border_th">Name</chef-th>
          <chef-th class="no_border_th">Description</chef-th>
          <chef-th class="no_border_th">Environment</chef-th>
```

File: components/automate-ui/src/app/modules/infra-proxy/policy-files/policy-files.component.html [hunk 7]
```diff
@@ -4,9 +4,9 @@
   <ng-container *ngIf="!policyFilesListLoading && !authFailure">
     <chef-table>
       <chef-thead>
-        <chef-tr>
-          <chef-th>Policy Name</chef-th>
-          <chef-th>Policy Group</chef-th>
+        <chef-tr class="no_border_tr">
+          <chef-th class="no_border_th">Name of Policies</chef-th>
+          <chef-th class="no_border_th">Group of Policies </chef-th>
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/policy-files/policy-files.component.html:9]:
```suggestion
          <chef-th class="no_border_th">Name</chef-th>
          <chef-th class="no_border_th">Group</chef-th>
```

File: components/automate-ui/src/app/modules/infra-proxy/policy-files/policy-files.component.html [hunk 8]
```diff
@@ -4,9 +4,9 @@
   <ng-container *ngIf="!policyFilesListLoading && !authFailure">
     <chef-table>
       <chef-thead>
-        <chef-tr>
-          <chef-th>Policy Name</chef-th>
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/policy-files/policy-files.component.html:8]: Neither your PR preamble nor the associated issue mentions this policy page--so is this really supposed to be included?

**Review by msorens — CHANGES_REQUESTED** [review]
There is apparently no UX specification for this issue and PR, so I will make my recommendation: let's omit the "of x" for every header.
It does not add anything to use headers Name of Role, Description of Role, Environment of Role; it only makes it cluttered.
See my specific comments below.
Note that I have also made all such terms singular; previously it was inconsistent with some singular and some plural.

File: components/automate-ui/src/app/modules/infra-proxy/policy-files/policy-files.component.html (hunk 8 above)
Comment (vinay033) [reply components/automate-ui/src/app/modules/infra-proxy/policy-files/policy-files.component.html:8]: yes for now.

**Comment by vinay033** [comment]
> There is apparently no UX specification for this issue and PR, so I will make my recommendation: let's omit the "of x" for every header.
> It does not add anything to use headers Name of Role, Description of Role, Environment of Role; it only makes it cluttered.
> See my specific comments below.
> Note that I have also made all such terms singular; previously it was inconsistent with some singular and some plural.
I did changes as per the UX suggestions but now I have changed it to `Name` only and updated the PR.

**Comment by kalroy** [comment]
> @kalroy Why did you mark this PR blocked?

I thought it was blocked by an API PR waiting to be merged. Since it is not the case, I am moving it out to Review.

**Comment by chaitali-mane** [comment]
While testing found minor CSS issues like more column width in list data, More space for list heading.

**Comment by msorens** [comment]
@chaitali-mane Please create a new GitHub issue for the CSS issues you indicated.

**Comment by chaitali-mane** [comment]
> @chaitali-mane Please create a new GitHub issue for the CSS issues you indicated.

Created issue for the UI chages #4750 and raised PR with changes #4754 against issue.

**Pull request merged**